app/screens/browse_channels/browse_channels.tsx

```diff
@@ -0,0 +1,501 @@
+import {ARCHIVED, PUBLIC, SHARED} from './constants';
+
+import type MyChannelModel from '@typings/database/models/servers/my_channel';
+
+type Props = {
+
+    // Screen Props (do not change during the lifetime of the screen)
+    componentId: string;
+    categoryId?: string;
+    closeButton: ImageResource;
+
+    // Calculated Props
+    canCreateChannels: boolean;
+    currentUserId: string;
+    currentTeamId: string;
+    joinedChannels?: MyChannelModel[];
+    sharedChannelsEnabled?: boolean;
+    canShowArchivedChannels: boolean;
+}
+
+const filterChannelsByTerm = (channels: Channel[], term: string) => {
+    const lowerCasedTerm = term.toLowerCase();
+    return channels.filter((c) => {
+        return (c.name.toLowerCase().includes(lowerCasedTerm) || c.display_name.toLowerCase().includes(lowerCasedTerm));
+    });
+};
+
+const filterJoinedChannels = (joinedChannels: MyChannelModel[], allChannels: Channel[] | undefined) => {
+    const ids = joinedChannels.map((c) => c.id);
+    return allChannels?.filter((c) => !ids.includes(c.id));
+};
+
+const makeLeftButton = (icon: ImageResource): OptionsTopBarButton => {
+    return {
+        id: 'close-more-channels',
+        icon,
+        testID: 'close.more_channels.button',
+    };
+};
+
+const makeRightButton = (theme: Theme, formatMessage: IntlShape['formatMessage']): OptionsTopBarButton => {
+    return {
+        color: theme.sidebarHeaderTextColor,
+        id: 'create-pub-channel',
+        text: formatMessage({id: 'mobile.create_channel', defaultMessage: 'Create'}),
+        showAsAction: 'always',
+        testID: 'more_channels.create.button',
+    };
+};
+
+const close = () => {
+    Keyboard.dismiss();
+    dismissModal();
+};
+
+const channelKeyExtractor = (channel: Channel) => {
+    return channel.id;
+};
+const getStyleFromTheme = makeStyleSheetFromTheme((theme: Theme) => {
+    return {
+        container: {
+            flex: 1,
+        },
+        searchBar: {
+            marginHorizontal: 12,
+            borderRadius: 8,
+            marginTop: 12,
+            backgroundColor: changeOpacity(theme.centerChannelColor, 0.08),
+        },
+        noResultContainer: {
+            flexGrow: 1,
+            alignItems: 'center',
+            justifyContent: 'center',
+        },
+        noResultText: {
+            fontSize: 26,
+            color: changeOpacity(theme.centerChannelColor, 0.5),
+        },
+        searchBarInput: {
+            color: theme.centerChannelColor,
+        },
+        loadingContainer: {
+            flex: 1,
+            justifyContent: 'center',
+            alignItems: 'center',
+        },
+        loading: {
+            height: 32,
+            width: 32,
+            justifyContent: 'center',
+        },
+        listContainer: {
+            paddingHorizontal: 20,
+            flexGrow: 1,
+        },
+    };
+});
+
+export default function BrowseChannels({
+    componentId,
+    canCreateChannels = false,
```

Comment:
why optional with a default to false?

Reply:
same with the others

Reply:
No real reason. At some point was marked as optional in V1 probably, and when it started complaining, I went with defaults for everyone :sweat_smile: Fixed.